hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALSplitter.java

```diff
@@ -1565,49 +1569,23 @@ Path closeWriter(String encodedRegionName, WriterAndPath wap, List<IOException>
 
 
     private List<IOException> closeLogWriters(List<IOException> thrown) throws IOException {
-      if (writersClosed) {
-        return thrown;
-      }
-
       if (thrown == null) {
         thrown = Lists.newArrayList();
       }
-      try {
-        for (WriterThread t : writerThreads) {
-          while (t.isAlive()) {
-            t.shouldStop = true;
-            t.interrupt();
-            try {
-              t.join(10);
-            } catch (InterruptedException e) {
-              IOException iie = new InterruptedIOException();
-              iie.initCause(e);
-              throw iie;
-            }
-          }
-        }
-      } finally {
-        WriterAndPath wap = null;
-        for (SinkWriter tmpWAP : writers.values()) {
+
+      for (WriterThread t : writerThreads) {
+        while (t.isAlive()) {
+          t.shouldStop = true;
+          t.interrupt();
           try {
-            wap = (WriterAndPath) tmpWAP;
-            wap.w.close();
```

Comment:
We are already closing the log files via WALSPlitter#close. Don't know the context why we are closing it here also.

Reply:
That's correct. @saintstack @ramkrish86 any thoughts about this?